lms/djangoapps/course_api/blocks/transformers/tests/test_milestones.py

```diff
@@ -223,3 +225,31 @@ def get_blocks_and_check_against_expected(self, user, expected_blocks):
             self.transformers,
         )
         assert set(block_structure.get_block_keys()) == set(self.get_block_key_set(self.blocks, *expected_blocks))
+
+    @override_waffle_flag(EXAMS_IDA, active=False)
+    @patch('lms.djangoapps.course_api.blocks.transformers.milestones.get_attempt_status_summary')
+    def test_exams_ida_flag_off(self, mock_get_attempt_status_summary):
+        self.transformers = BlockStructureTransformers([self.TRANSFORMER_CLASS_TO_TEST(True)])
+        self.course.enable_subsection_gating = True
+        self.setup_gated_section(self.blocks['H'], self.blocks['A'])
+        expected_blocks = (
+            'course', 'A', 'B', 'C', 'ProctoredExam', 'D', 'E', 'PracticeExam', 'F', 'G', 'TimedExam', 'J', 'K', 'H',
+            'I')  # lint-amnesty, pylint: disable=line-too-long
+        self.get_blocks_and_check_against_expected(self.user, expected_blocks)
+
+        # Ensure that call is made to get_attempt_status_summary
+        assert mock_get_attempt_status_summary.call_count > 0
+
+    @override_waffle_flag(EXAMS_IDA, active=True)
+    @patch('lms.djangoapps.course_api.blocks.transformers.milestones.get_attempt_status_summary')
+    def test_exams_ida_flag_on(self, mock_get_attempt_status_summary):
```

Comment:
If I also wanted to test the description when the waffle flag is on, should I just test the helper method? Or is it better practice to test the main method that uses the helper method?